ocaml/typing/typeopt.ml

```diff
@@ -61,7 +61,7 @@ let scrape_poly env ty =
   | d -> d
 
 let is_function_type env ty =
-  match scrape env ty with
+  match scrape_poly env ty with
```

Comment:
While this change seems completely fine, I'm struggling to see why it's needed.  I wouldn't expect `TPoly` nodes to occur in the relevant places, and the tests seem to pass if I undo it.

Reply:
I don't seem to be able to reproduce the failures I had previously, it might be related to some other changes (the ocaml-jst merge, maybe?). In any case, I reverted this change.